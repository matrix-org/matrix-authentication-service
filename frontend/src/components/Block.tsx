// Copyright 2022 The Matrix.org Foundation C.I.C.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

type Props = {
  children: React.ReactNode;
  highlight?: boolean;
  className?: string;
};

const Block: React.FC<Props> = ({ children, highlight, className }) => {
  return (
    <div
      className={`p-4 dark:text-white grid rounded ${className} ${
        highlight
          ? "border-2 border-grey-50 dark:border-grey-450 bg-white dark:bg-black"
          : "bg-grey-50 dark:bg-grey-450"
      }`}
    >
      {children}
    </div>
  );
};

export default Block;
