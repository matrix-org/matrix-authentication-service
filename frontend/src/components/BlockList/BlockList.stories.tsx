// Copyright 2023 The Matrix.org Foundation C.I.C.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { Meta, StoryObj } from "@storybook/react";
import { H2, Text } from "@vector-im/compound-web";

import Block from "../Block";

import BlockList from "./BlockList";

const meta = {
  title: "UI/Block List",
  component: BlockList,
} satisfies Meta<typeof BlockList>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Basic: Story = {
  render: (args) => (
    <BlockList {...args}>
      <Block>
        <H2>Block 1</H2>
        <Text>Body 1</Text>
      </Block>
      <Block>
        <H2>Block 2</H2>
        <Text>Body 2</Text>
      </Block>
    </BlockList>
  ),
};
