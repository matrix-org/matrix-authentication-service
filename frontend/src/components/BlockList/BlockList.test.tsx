// Copyright 2023 The Matrix.org Foundation C.I.C.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// @vitest-environment happy-dom

import { create } from "react-test-renderer";
import { describe, expect, it } from "vitest";

import Block from "../Block";

import BlockList from "./BlockList";

describe("BlockList", () => {
  it("render an empty <BlockList />", () => {
    const component = create(<BlockList />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("render <BlockList /> with children", () => {
    const component = create(
      <BlockList>
        <Block>Block 1</Block>
        <Block>Block 2</Block>
      </BlockList>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("passes down the className prop", () => {
    const component = create(<BlockList className="foo" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
