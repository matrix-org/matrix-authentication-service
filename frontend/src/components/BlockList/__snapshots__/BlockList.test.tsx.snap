// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlockList > passes down the className prop 1`] = `
<div
  className="_blockList_f8cc7f foo"
/>
`;

exports[`BlockList > render <BlockList /> with children 1`] = `
<div
  className="_blockList_f8cc7f"
>
  <div
    className="_block_17898c"
  >
    Block 1
  </div>
  <div
    className="_block_17898c"
  >
    Block 2
  </div>
</div>
`;

exports[`BlockList > render an empty <BlockList /> 1`] = `
<div
  className="_blockList_f8cc7f"
/>
`;
